packages/peregrine/lib/talons/CartPage/ProductListing/useProduct.js

```diff
@@ -0,0 +1,80 @@
+import { useCallback, useState } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+
+import { useCartContext } from '../../../context/cart';
+
+export const useProduct = props => {
+    const {
+        item,
+        refetchCartQuery,
+        refetchPriceQuery,
+        removeItemMutation
+    } = props;
+
+    const flatProduct = flattenProduct(item);
+    const [removeItem] = useMutation(removeItemMutation);
+
+    const [{ cartId }] = useCartContext();
+
+    const [isRemoving, setIsRemoving] = useState(false);
+    const [isFavorite, setIsFavorite] = useState(false);
+
+    const handleToggleFavorites = useCallback(() => {
+        setIsFavorite(!isFavorite);
+    }, [isFavorite]);
+
+    const handleEditItem = useCallback(() => {
+        // Edit Item action to be completed by PWA-272.
+    }, []);
+
+    const handleRemoveFromCart = useCallback(async () => {
+        setIsRemoving(true);
```

Comment:
You . may not have to use a local state. The `removeItem` mutation returns an object with a `loading` prop. I'm not sure if it waits for refetchQueries though.